src/main/java/frc/robot/subsystems/VisionSubsystem.java

```diff
@@ -47,225 +49,32 @@
-  }
-
-  public Matrix<N3, N1> getCurrentStdDevs(){
-    return curStdDevs;
+  public ArrayList<LocalizationCamera> getLocalizationCameras(){
+    return cameras;
   }
 
   @Override
   public void periodic() {
-
-    double targetDistanceMeters = 0.0;
-
-    var results = m_camera.getAllUnreadResults();
-
-    ArrayList<Integer> targetIds = new ArrayList<Integer>();
-    ArrayList<Double> targetPoseAmbiguity = new ArrayList<Double>();
-
-    if (!results.isEmpty()) {
-      // Camera processed a new frame since last
-      // Get the last one in the list.
-      var result = results.get(results.size() - 1);
-
-      if (result.hasTargets()) {
-        PhotonTrackedTarget target = null;
-        double biggestTargetArea = 0;
-
-        for (PhotonTrackedTarget sampleTarget : result.getTargets()){
-          targetIds.add(sampleTarget.getFiducialId());
-          targetPoseAmbiguity.add(sampleTarget.getPoseAmbiguity());
-          //loops through every sample target in results.getTargets()
-          //if the sample target's area is bigger than the biggestTargetArea, then the sample target
-          // is set to the target, and the biggest Target Area is set to the sample's target area
-          if (sampleTarget.getArea() > biggestTargetArea){
-            biggestTargetArea = sampleTarget.getArea();
-            target = sampleTarget;
-          }
-
-        }
-
-        if (target.getPoseAmbiguity() > VisionConstants.MAX_POSE_AMBIGUITY) {
-          SmartDashboard.putString("vision/" + m_cameraName + "/targetState", "targetDiscardedAmbiguity");
-          targetFound = false;
-        } else {
-          SmartDashboard.putString("vision/" + m_cameraName + "/targetState", "targetFound");
-          targetFound = true;
-        }
-        
-        targetYaw = aprilTagFieldLayout.getTagPose(target.getFiducialId()).get().getRotation().getZ();
-        targetPose = aprilTagFieldLayout.getTagPose(target.getFiducialId()).get().toPose2d();
-      } else {
-        SmartDashboard.putString("vision/" + m_cameraName + "/targetState", "noTarget");
-        targetFound = false;
-      }
-
-      if (targetFound) {
-        // update the estimated robot pose if the pose estimator outputs something
-        Optional<EstimatedRobotPose> poseEstimatorOutput = poseEstimator.update(result);
-
-        // update standard deviation based on dist 
-        this.updateEstimationStdDevs(poseEstimatorOutput, result.getTargets());
-
-        if (poseEstimatorOutput.isPresent() && VisionUtils.poseIsSane(poseEstimatorOutput.get().estimatedPose)) {
-          estimatedRobotPose = poseEstimatorOutput.get(); 
-
-          // update our last n poses
-          lastEstPoses.add(estimatedRobotPose.estimatedPose.toPose2d());
-          if (lastEstPoses.size() > VisionConstants.NUM_LAST_EST_POSES) {
-            lastEstPoses.remove(0);
-          }
-        }
+    for (LocalizationCamera cam : cameras){
+      cam.findTarget();
+      
+      if (cam.getTargetFound()){
```

Comment:
I don't think you need this at all. A lot of the "target" logic was from before we were using vision for pose estimation. But you are adding elements to a list and never removing them, so at the very least you'll have to clear the list at the beginning of periodic.